packages/guides/src/Compiler/Passes/FooternoteNumberedTargetPass.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\Passes;
+
+use phpDocumentor\Guides\Compiler\CompilerPass;
+use phpDocumentor\Guides\Meta\FootnoteTarget;
+use phpDocumentor\Guides\Metas;
+use phpDocumentor\Guides\Nodes\CompoundNode;
+use phpDocumentor\Guides\Nodes\FootnoteNode;
+use phpDocumentor\Guides\Nodes\Node;
+
+/**
+ * Resolves the hyperlink target for each section in the document.
+ *
+ * This follows the reStructuredText rules as outlined in:
+ * https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#implicit-hyperlink-targets
+ */
+class FooternoteNumberedTargetPass implements CompilerPass
+{
+    public function __construct(private readonly Metas $metas)
+    {
+    }
+
+    public function getPriority(): int
+    {
+        return 30000; // must be run *before* FooternoteNamedTargetPass
+    }
+
+    /** {@inheritDoc} */
+    public function run(array $documents): array
+    {
+        foreach ($documents as $document) {
+            $this->traverse($document, $document->getFilePath());
+        }
+
+        return $documents;
+    }
+
+    private function traverse(Node $node, string $path): void
+    {
+        if (!($node instanceof CompoundNode)) {
+            return;
+        }
+
+        foreach ($node->getChildren() as $child) {
+            if ($child instanceof FootnoteNode && $child->getNumber() > 0) {
+                $this->metas->addFootnoteTarget(new FootnoteTarget(
+                    $path,
+                    $child->getAnchor(),
+                    '',
+                    $child->getNumber(),
+                ));
+            } elseif ($child instanceof CompoundNode) {
+                $this->traverse($child, $path);
+            }
```

Comment:
This makes it a bit easier to read. We want to add the recursion, and take the payload, 

Or create a new Footnote target

```suggestion
            if ($child instanceof CompoundNode) {
                $this->traverse($child, $path);
                continue;
            }
            
            if ($child instanceof FootnoteNode && $child->getNumber() > 0) {
                $this->metas->addFootnoteTarget(new FootnoteTarget(
                    $path,
                    $child->getAnchor(),
                    '',
                    $child->getNumber(),
                ));
            } 
```